subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/gcc/ArStaticLibraryArchiver.java

```diff
@@ -74,7 +75,9 @@ public void execute(BuildOperationQueue<CommandLineToolInvocation> buildQueue) {
 
     @Override
     protected void addOptionsFileArgs(List<String> args, File tempDir) {
-        // No support for command file
```

Comment:
❓I would double check with Git blame to see if there is more information as to why there were no support? I'm wondering if it was due to an older GCC version. If I'm remembering correctly, the minimum version was 4.x back in the days. It may be safe to change now to unblock users. Nokee (https://nokee.dev) will take a bit of a different approach to allow for easy configuration of this behavior.

Reply:
I'm afraid I couldn't find any information on why command files are not activated for `ar`.
The static library archiver was introduced in commit 1b10c37 without support for command files.
This particular comment was added in commit 659e1a2 during the work on the initial support for Swift.
I think command files are generally disabled for gcc versions prior to 4.0 since commit 2760cc6:
https://github.com/gradle/gradle/blob/9be34206974bcdf71f1ce7876cbf9215f8d8e8c1/subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/gcc/GccToolChain.java#L50

GNU `ar` is not part of gcc, but GNU binutils and thus has different versioning than gcc.
According to its [news](https://sourceware.org/git/?p=binutils-gdb.git;a=blob;f=binutils/NEWS;hb=binutils-2_37#l489), the `ar` tool of GNU binutils seems to support the `@file` syntax since version 2.17 which has been released in 2006.
However, I don't know whether other variants of `ar` support command files and if we have to consider them as well?
For example command files for assemblers appear to be disabled since commit 82fd940 because some custom assemblers don't support them.

I'm looking forward to see the approach of Nokee on tool chain configuration.